apps/restconf/restconf_methods.c

```diff
@@ -581,6 +581,367 @@ api_data_write(clicon_handle h,
 } /* api_data_write */
 
 #ifdef YANG_PATCH
+
+char * init_str() 
```

Comment:
if you need safe, resizeable vector, you probably can use cligen's cbuf

Reply:
I changed the code to use cbufs, as you requested, but Olof wrote me today after reviewing my code that he prefers malloc functions to cbuf. In his words, "it seems the use of cbufs make the code more complex. Cbufs were not meant as a replacement of string functions (eg strcmp etc), just for buffer handling." So how should I handle this?

Reply:
Sorry, no that was not really what I said, or at least not what I meant. 
I support the use of cbufs, and not defining a string-malloc-lib specific to the file.
My concern is that the buffer code seems overly complex, and was asking whether there is a possibility to make it simpler?
One problem is it seems geared towards JSON, but it need to handle XML as well.
One could try to work with the internal xml structure: cxobj:s and the translate that to XML or JSON, instead.